### PR TITLE
Incorporate Sven's and Louis' comments to the release notes.

### PR DESCRIPTION
**Pull request by geeknoid**

### CODE REVIEW

File: _docs/reference/release-notes.md [hunk 1]
```diff
@@ -46,21 +47,23 @@ request-time for initial requests and thus delivers a smoother average latency.
 
 - **New Protocols**. Mesh-wide support for WebSocket connections, MongoDB proxying.
 
-- **Other Improvements**. Ingress properly supports gRPC services, interfacing headless services, better support for health checks, and
```
Comment (linsun) [inline _docs/reference/release-notes.md]: curious why is headless service removed?

**Comment by geeknoid** [comment]
It wasn't clear to Sven and myself what this meant. Do you have a different
phrasing for this perhaps?

**Comment by linsun** [comment]
@ijsnellf added support for headless service so the pods can be k8s headless service:  https://kubernetes.io/docs/concepts/services-networking/service/#headless-services

A lot of users asked for this on the mailing list.

**Comment by geeknoid** [comment]
Added back, with a link to k8s. Thanks.

**Pull request merged**